### ISSUE TITLE
minetest.place_node() causes exception in default.log_player_action()

### ISSUE THREAD

**Issue by bengardner** [minetest/minetest_game#2993]
The working_villagers mod uses ```minetest.place_node()``` to plant saplings, for example, "default:sapling".

When placed, it calls ```default.sapling_on_place()```, which calls ```default.log_player_action(placer, "places node", sapling_name, "at", pos)```.  When the "on_place" callback is triggered via ```place_node()```, the ```placer``` is nil.

The log_player_action() derefs the nil player and crashes.
```lua
function default.log_player_action(player, ...)
	local msg = player:get_player_name()
	...
end
```

A simple fix is to add a guard to protect against player == nil.
```lua
function default.log_player_action(player, ...)
	if player == nil then
		return
	end
	local msg = player:get_player_name()
	...
end
```
I've looked through the c++ code (ModApiEnvMod::l_place_node() and ScriptApiItem::item_OnPlace()) and it looks like the player will always be null.
This appears to have been introduced with https://github.com/minetest/minetest_game/pull/2941

See https://github.com/theFox6/working_villages/issues/32 for more details in reproducing the issue.

**Comment by appgurueu** [comment]
Yes, `placer` may be `nil` and this was missed in the PR. I'd fix it in the sapling `on_place` rather than `default.log_player_action` though; other `on_place`s don't seem to have the error.

**Comment by bengardner** [comment]
I see two other possible crash points related to ```default.log_player_action```.

```default.coral_on_place()``` would crash if a coral is placed via ```minetest.place_node()```.
placer is used without checking. Same as the sapling.

```farming.place_seed()``` also might crash. It checks for a nil placer, but passes placer to log_player_action() without checking for nil. The nil check implies that placer might be nil.

**Comment by appgurueu** [comment]
> `default.coral_on_place()` would crash if a coral is placed via `minetest.place_node()`. placer is used without checking. Same as the sapling.

There is no `default.coral_on_place()`:

```
$ grep -r coral_on_place
mods/default/nodes.lua:local function coral_on_place(itemstack, placer, pointed_thing)
mods/default/nodes.lua:	on_place = coral_on_place,
mods/default/nodes.lua:	on_place = coral_on_place,
mods/default/nodes.lua:	on_place = coral_on_place,
```

The local `coral_on_place` seems to be fine:

```lua
local function coral_on_place(itemstack, placer, pointed_thing)
	if pointed_thing.type ~= "node" or not placer then
		return itemstack
	end
```

> `farming.place_seed()` also might crash. It checks for a nil placer, but passes placer to log_player_action() without checking for nil.

You're right there.

> The nil check implies that placer might be nil.

You'd be surprised how many unnecessary `nil` checks there are :P
